pkg/fn/eval.go

```diff
@@ -17,13 +17,18 @@ package fn
 import (
 	"context"
 
+	fnresult "github.com/GoogleContainerTools/kpt/pkg/api/fnresult/v1"
 	v1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
 	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/kustomize/kyaml/kio"
 )
 
 type EvalOptions struct {
+	// FnResultList stores the result of the function evaluation
+	FnResultList *fnresult.ResultList
 }
 
 type Evaluator interface {
 	Eval(ctx context.Context, pkg filesys.FileSystem, fn v1.Function, opts EvalOptions) error
+	NewRunner(ctx context.Context, fn *v1.Function, opts EvalOptions) (kio.Filter, error)
```

Comment:
Couple suggestions:

NewRunner --> NewFilter
EvalOptions --> FilterOptions

Reply:
FilterOptions seems generic. NewRunner is consistent with fnruntime.

Reply:
If we are going to stick with the fnruntime nomenclature, let's switch over completely:

Evaluator --> FunctionRunner
eval.go --> runner.go
EvalOptions --> RunnerOptions

wdyt?

I still wish NewRunner returned Runner and not Filter, but oh well :)